File: core/src/main/java/com/core/adapter/AbstractSimpleHolder.java
package com.core.adapter;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import android.view.View;

/**
 * @author volkanhotur
 */

public abstract class AbstractSimpleHolder<Data> extends RecyclerView.ViewHolder {
    public AbstractSimpleHolder(@NonNull View itemView) {
        super(itemView);
    }

    abstract void bind(Data data);
}

File: core/src/main/java/com/core/adapter/AbstractSimpleHolder.kt
package com.core.adapter

import android.view.View
import androidx.recyclerview.widget.RecyclerView

/**
 * @author volkanhotur
 */
abstract class AbstractSimpleHolder<Data>(itemView: View) : RecyclerView.ViewHolder(itemView) {

    abstract fun bind(data: Data)
}